analysis/tools/ccsh/src/main/kotlin/de/maibornwolff/codecharta/tools/ccsh/Ccsh.kt

```diff
@@ -75,15 +87,80 @@ class Ccsh : Callable<Void?> {
         fun executeCommandLine(args: Array<String>): Int {
             val commandLine = CommandLine(Ccsh())
             commandLine.executionStrategy = CommandLine.RunAll()
-            if (args.isEmpty() || isParserUnknown(args, commandLine)) {
-                return executeInteractiveParser(commandLine)
+            return if (args.isEmpty()) {
+                val configuredParsers = offerInteractiveParserSuggestions(commandLine)
+                if (configuredParsers.isEmpty()) {
+                    return 0
+                }
+
+                return executeConfiguredParsers(commandLine, configuredParsers)
+            } else if (isParserUnknown(args, commandLine) || args.contains("--interactive") || args.contains("-i")) {
+                executeInteractiveParser(commandLine)
             } else {
-                return commandLine.execute(*sanitizeArgs(args))
+                commandLine.execute(*sanitizeArgs(args))
             }
         }
 
+        fun offerInteractiveParserSuggestions(commandLine: CommandLine): Map<String, List<String>> {
+            val inputFile: String = KInquirer.promptInput(
+                    message = "What is the name of the input file/folder/url?",
+                    hint = "path/to/input/")
+
+            val selectedParsers = ParserService.offerParserSuggestions(commandLine, PicocliParserRepository(), inputFile)
+
+            if (selectedParsers.isEmpty()) {
+                println(NO_USABLE_PARSER_FOUND_MESSAGE)
+                return emptyMap()
+            }
+
+            val shouldRunConfiguredParsers: Boolean =
+                    KInquirer.promptConfirm(
+                            message = "Do you want to run all configured parsers?",
+                            default = true)
+
+            if (!shouldRunConfiguredParsers) {
+                return emptyMap()
+            }
+
+            return (ParserService.configureParserSelection(commandLine, PicocliParserRepository(), selectedParsers))
+        }
+
+        fun executeConfiguredParsers(commandLine: CommandLine, configuredParsers: Map<String, List<String>>): Int {
+            val exitCode = AtomicInteger(0)
+            val threadPool = Executors.newFixedThreadPool(configuredParsers.size)
```

Comment:
Is there a way to calculate an upper bound of threads to be started?

Reply:
Good idea, for now I decided against putting more effort into this because the number of unique parsers is so low.
If we implemented all parsers to be suggestable, we would have 15 or something parsers. Even if we doubled that amount somehow - is starting 30 threads too much?

Reply:
As we decided we will limit the amount of threads that we start by the following formula:
min(parsersToBeStarted, CPU-Cores)